src/pages/en/core-concepts/routing.md

```diff
@@ -19,65 +19,93 @@ src/pages/about/me.astro     -> mysite.com/about/me
 src/pages/posts/1.md         -> mysite.com/posts/1
 ```
 
+> 💡 There is no separate "routing config" to maintain in an Astro project. Static pages are created by placing files in the `/src/pages/` directory.
+
 ## Dynamic routes
 
-Sometimes, you need to generate many URLs from a single page component. Astro uses file-based routing to support **dynamic route parameters** in the filename, so that one page can match many dynamic routes based on some pattern.
+A single Astro Page component can also specify dynamic route parameters in its filename to generate multiple routes that match a given criteria. You can create several related pages at once, such as author pages, or a page for each blog tag. Named parameters allow you to provide multiple values for "named" levels of these route paths
+
+> 💡 Even dynamically-created pages and routes are generated at build time.
 
-An important thing to keep in mind: Astro is a static site builder. There is no Astro server to run in production, which means that every page must be built ahead of time. Pages that use dynamic routes must export a `getStaticPaths()` function which will tell Astro exactly what pages to generate. Learn more by viewing the complete [API Reference](/en/reference/api-reference#getstaticpaths).
+Astro pages that create dynamic routes must: 
 
-### Named parameters
+1. use `[bracket]` notation to identify the dynamic parameters
 
-Dynamic parameters are encoded into the filename using `[bracket]` notation:
+2. export a `getStaticPaths()` function to specify exactly which paths will be pre-rendered by Astro.
+
+```astro
+---
+// src/pages/dogs/[dog].astro
+
+export function getStaticPaths() {
+  return [
+    // Generates: /dogs/clifford
+    {params: {dog: 'clifford'}},
+    // Generates: /dogs/rover
+    {params: {dog: 'rover'}},
+    // Generates: /dogs/spot
+    {params: {dog: 'spot'}},
+  ];
+}
+---
+```
 
-- `pages/blog/[slug].astro` → `/blog/:slug` (`/blog/hello-world`, `/blog/post-2`, etc.)
+📚 Learn more about [`getStaticPaths()`](/en/reference/api-reference#getstaticpaths).
+
+Routes can be generated from multiple named parameters, at any level of the filepath:
+
+- `pages/blog/[slug].astro` → (`/blog/hello-world`, `/blog/post-2`, etc.)
 - `pages/[username]/settings.astro` → (`/fred/settings`, `/drew/settings`, etc.)
 - `pages/[lang]-[version]/info.astro` → (`/en-v1/info`, `/fr-v2/info`, etc.)
 
-#### Example: Named parameters
+#### Requesting the parameters
```

Comment:
"requesting" has a special meaning, I thought that you were talking about making a separate HTTP request. Maybe "Reading" or "Accessing" instead?

Reply:
Or, maybe something more like the previous header: `## Named Parameters`

Reply:
What about using Astro.request.params as the header itself? Since that might be something people might want to see/skim for in the right hand sidebar? And, I'd nest it under Dynamic Routing with three # so the sidebar would be:

- Static Routes
- Dynamic Routes
- - Astro.request.params
- - Rest parameters
- - Caveats

Reply:
If you have `Rest Parameters`, you need `Named Parameters` imo since they're the two different ways to write your params.

If you wanted to dedicate a section just to accessing params in the frontmatter/template via `Astro.request.params`, you could give that its own section as well, but I don't think its required imo.

Reply:
OK, will bring the header back and rework.

I had the picture in my head that for dynamic routes at all you NEED named parameters, and adding the rest is like an "add-on" to the name, so rest parameters didn't seem like a different thing entirely, just a more highly specialized version of it.

But I'll rework to make them clearly, as you say, two different ways to write the params!